src/test/enhetstester/kotlin/no/nav/familie/ba/sak/kjerne/autovedtak/svalbardtillegg/AutovedtakSvalbardtilleggBegrunnelseServiceTest.kt.kt

```diff
@@ -0,0 +1,251 @@
+    private val beregningService = mockk<BeregningService>()
+    private val vedtaksperiodeService = mockk<VedtaksperiodeService>()
+    private val vedtakService = mockk<VedtakService>()
+    private val vedtaksperiodeHentOgPersisterService = mockk<VedtaksperiodeHentOgPersisterService>()
+
+    private val autovedtakSvalbardtilleggBegrunnelseService =
+        AutovedtakSvalbardtilleggBegrunnelseService(
+            behandlingHentOgPersisterService = behandlingHentOgPersisterService,
+            beregningService = beregningService,
+            vedtaksperiodeService = vedtaksperiodeService,
+            vedtakService = vedtakService,
+            vedtaksperiodeHentOgPersisterService = vedtaksperiodeHentOgPersisterService,
+        )
+
+    private val fagsak = defaultFagsak()
+    private val forrigeBehandling = lagBehandling(fagsak = fagsak, id = 0)
+    private val behandling = lagBehandling(fagsak = fagsak, id = 1)
+    private val barn = lagPerson(type = PersonType.BARN)
+    private val barn2 = lagPerson(type = PersonType.BARN)
+
+    @Test
+    fun `Skal kaste feil dersom det ikke finnes noen perioder å begrunne`() {
+        // Arrange
+        val forrigeAndeler =
+            listOf(
+                lagAndelTilkjentYtelse(
+                    fom = YearMonth.of(2025, 11),
+                    tom = YearMonth.of(2025, 12),
+                    person = barn,
+                    behandling = forrigeBehandling,
+                    beløp = 500,
+                    sats = 500,
+                    ytelseType = YtelseType.SVALBARDTILLEGG,
+                ),
+            )
+
+        val nåværendeAndeler =
+            listOf(
+                lagAndelTilkjentYtelse(
+                    fom = YearMonth.of(2025, 11),
+                    tom = YearMonth.of(2025, 12),
+                    person = barn,
+                    behandling = behandling,
+                    beløp = 500,
+                    sats = 500,
+                    ytelseType = YtelseType.SVALBARDTILLEGG,
+                ),
+            )
+
+        every { behandlingHentOgPersisterService.hentSisteBehandlingSomErIverksatt(fagsak.id) } returns forrigeBehandling
+        every { beregningService.hentAndelerTilkjentYtelseMedUtbetalingerForBehandling(forrigeBehandling.id) } returns forrigeAndeler
+        every { beregningService.hentAndelerTilkjentYtelseMedUtbetalingerForBehandling(behandling.id) } returns nåværendeAndeler
+
+        // Act && Assert
+        val feilmelding =
+            assertThrows<Feil> {
+                autovedtakSvalbardtilleggBegrunnelseService.begrunnAutovedtakForSvalbardtillegg(behandling)
+            }.message
+
+        assertThat(feilmelding).isEqualTo("Det er forsøkt å begrunne autovedtak men det ble ikke funnet noen perioder med innvilgelse eller reduksjon.")
+    }
+
+    @Test
+    fun `Skal lagre vedtaksperiode med innvilgesebegrunnelse dersom det finnes perioder med innvilgelse`() {
```

Comment:
Nitpick, men endre det gjerne i filen for finnmarkstilleggbehandlinger også
```suggestion
    fun `Skal lagre vedtaksperiode med innvilgelsebegrunnelse dersom det finnes perioder med innvilgelse`() {
```